composer.json

```diff
@@ -11,16 +11,17 @@
         }
     ],
     "require": {
-        "php": "^5.4|7.*",
-        "puli/composer-plugin": "1.0.0-beta9"
+        "php": "^5.4|7.*"
     },
     "require-dev": {
         "php-http/httplug": "^1.0",
         "php-http/message-factory": "^1.0",
+        "puli/composer-plugin": "1.0.0-beta9",
         "phpspec/phpspec": "^2.4",
         "henrikbjorn/phpspec-code-coverage" : "^1.0"
     },
     "suggest": {
+        "puli/composer-plugin": "Allow to use Discovery",
```

Comment:
As @webmozart pointed out, puli composer plugin is not necessary to use puli. You can manually define your bindings, etc as well.

Basically you want to set up puli in your application or as a dev dependency. If you don't, and a request is made to the discovery layer, it will look for a puli setup and throw an exception if there isn't any. This makes Puli configuration your responsibility, while discovery comes as a dependency.

The point is: Puli can be set up multiple ways and it should be the user's choice how to do it.